app/scripts/controllers/attachPVC.js

```diff
@@ -175,24 +220,32 @@ angular.module('openshiftConsole')
             var readOnly = $scope.attach.readOnly;
             if (mountPath) {
               // for each container in the pod spec, add the new volume mount
-              angular.forEach(podTemplate.spec.containers, function(container) {
-                if (isContainerSelected(container)) {
-                  var newVolumeMount =
-                    StorageService.createVolumeMount(name, mountPath, subPath, readOnly);
-                  if (!container.volumeMounts) {
-                    container.volumeMounts = [];
-                  }
-                  container.volumeMounts.push(newVolumeMount);
-                }
-              });
+              // angular.forEach(podTemplate.spec.containers, function(container) {
+              if(!setVolumeMount(podTemplate, name, mountPath, subPath, readOnly)) {
+                $scope.disableInputs = false;
+                return;
+              }
             }
 
             // add the new volume to the pod template
             var newVolume = StorageService.createVolume(name, persistentVolumeClaim);
             if (!podTemplate.spec.volumes) {
               podTemplate.spec.volumes = [];
             }
-            podTemplate.spec.volumes.push(newVolume);
+
+            // if the newly created volume already exists, only
+            // fail if the "overwrite" option was not set
+            var volumeExists = false;
+            for (var i = 0; i < podTemplate.spec.volumes.length; i++) {
+              if (podTemplate.spec.volumes[i].name === newVolume.name) {
+                volumeExists = true;
+                break;
+              }
+            }
```

Comment:
You should use `_.some` to make this a one-liner:

```js
var volumeExists = _.some(podTemplate.spec.volumes, { name: newVolume.name });
```

https://lodash.com/docs/4.17.4#some